sql/catalyst/src/test/scala/org/apache/spark/sql/util/DateFormatterSuite.scala

```diff
@@ -169,19 +175,22 @@ class DateFormatterSuite extends SparkFunSuite with SQLHelper {
   test("SPARK-31557: rebasing in legacy formatters/parsers") {
     withSQLConf(SQLConf.LEGACY_TIME_PARSER_POLICY.key -> LegacyBehaviorPolicy.LEGACY.toString) {
       LegacyDateFormats.values.foreach { legacyFormat =>
-        outstandingTimezonesIds.foreach { timeZone =>
-          withSQLConf(SQLConf.SESSION_LOCAL_TIMEZONE.key -> timeZone) {
-            val formatter = DateFormatter(
-              DateFormatter.defaultPattern,
-              getZoneId(timeZone),
-              DateFormatter.defaultLocale,
-              legacyFormat,
-              isParsing = false)
-            assert(LocalDate.ofEpochDay(formatter.parse("1000-01-01")) === LocalDate.of(1000, 1, 1))
-            assert(formatter.format(LocalDate.of(1000, 1, 1)) === "1000-01-01")
-            assert(formatter.format(localDateToDays(LocalDate.of(1000, 1, 1))) === "1000-01-01")
-            assert(formatter.format(java.sql.Date.valueOf("1000-01-01")) === "1000-01-01")
-          }
+        withOutstandingZoneIds { zoneId =>
+          val formatter = DateFormatter(
+            DateFormatter.defaultPattern,
+            zoneId,
+            DateFormatter.defaultLocale,
+            legacyFormat,
+            isParsing = false)
+          assert(LocalDate.ofEpochDay(formatter.parse("1000-01-01")) === LocalDate.of(1000, 1, 1))
+          assert(formatter.format(LocalDate.of(1000, 1, 1)) === "1000-01-01")
+          assert(formatter.format(localDateToDays(LocalDate.of(1000, 1, 1))) === "1000-01-01")
+          val cal = new Calendar.Builder()
```

Comment:
The time zone is embedded into `java.sql.Date`, and it is the global default JVM time zone. To set tested time zone, I have to construct Date via the calendar.